routes/call.js

```diff
@@ -55,12 +55,19 @@ module.exports = function handleCallRequest(request, reply) {
   .then(res => res.json())
   .then(json => {
     if (json.error) {
-      throw new Error(json.error);
+      throw new Error(JSON.stringify(json));
     }
     reply({ 'call_placed': true }).code(200);
   })
   .catch(error => {
+    var error = error.message;
+    var status = 500;
+    try {
+      var data = JSON.parse(error);
+      error = data.error;
+      status = data.status;
+    } catch (e) { /* error message was not JSON data */ }
     console.error(`error for ${number}/${locale}/cid:${cid}(${parsed.country}):`, error);
-    reply({ 'call_placed': false, error: error }).code(500);
+    reply({ 'call_placed': false, error: error }).code(status);
```

Comment:
For scrutiny sake, if this is always a 500, I would rather it hard coded, gives no chance for a bug in the future where it's passing something else, because the logic around `status` changes but this line isn't noticed..

Reply:
If I'm reading this correctly, if there is cases where status isn't 500, then we can continue down the current road. :)

Reply:
status was hardcoded as 500, but that is not actually the case, this update changes it so that the API's status is forwarded instead of just saying "every error is a 500 error", which was blatantly wrong =D

Reply:
Ah, great. I think that's an acceptable reason.

Reply:
Looking at the code though, wouldn't it hit this line when it tries to do a `JSON.parse` meaning it never gets to `status=data.status`? Meaning it's always 500 still?

Reply:
In other words, it'll hit the catch before it sent status to anything but the 500? That's my understanding.

Reply:
if json.parse fails, then we have a genuine 500 error. But if it succeeds, we have access to the call-power API response status, and we should use that instead.